src/hooks/useConfirmDialog.js

```diff
@@ -0,0 +1,13 @@
+import { usePopup } from 'hooks'
+import { ConfirmDialog } from 'components'
+
+export const useConfirmDialog = () => {
+  const [showPopup] = usePopup(ConfirmDialog, { stack: true })
+  const showConfirmDialog = ({
+  	title, message, onDiscardText, onDiscard = () => {},
+  	onSubmitText, onSubmit = () => {}
+	}) => {
+		showPopup({ title, message, onDiscardText, onDiscard, onSubmitText, onSubmit })
+	}
+  return showConfirmDialog
+}
```

Comment:
👍 Nice! It also fixes the problem when I tried to update the preact to latest version due to the misuse of the hook not in the top-level component

```
hooks.module.js:1 Uncaught TypeError: Cannot read properties of undefined (reading 'context')
    at F (hooks.module.js:1)
    at usePopup (usePopup.js:5)
    at confirmDialog (confirm.js:8)
    at section (QuickFacts.js:37)
    at Object.onKeyCenter [as Enter] (useArticleLinksNavigation.js:104)
    at HTMLDocument.onKeyDown (Softkey.js:48)
```